src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -224,13 +224,20 @@ object LoadVCF {
       val hd = parseHeader(reader, getHeaderLines(hConf, file))
       val hd1 = header1Bc.value
 
-      hd1.sampleIds.iterator.zip(hd.sampleIds.iterator)
-        .zipWithIndex.dropWhile { case ((s1, s2), i) => s1 == s2 }.toArray.headOption match {
-        case Some(((s1, s2), i)) => fatal(
+      if (hd1.sampleIds.length != hd.sampleIds.length) {
+        fatal(
+          s"""invalid sample ids: sample ids are different lengths.
+             | ${files(0)} has ${hd1.sampleIds.length} ids and
+             | ${file} has ${hd.sampleIds.length} ids.
+           """.stripMargin)
+      }
+
+      hd1.sampleIds.iterator.zipAll(hd.sampleIds.iterator, None, None)
+        .zipWithIndex.foreach { case ((s1, s2), i) =>
+        if (s1 != s2) fatal(
```

Comment:
The fatal should go on a new line. There's an IntelliJ style file you should download (https://github.com/hail-is/hail/blob/master/code_style.xml). See directions here: https://github.com/hail-is/hail/blob/master/style-guide.md. Once IntelliJ has loaded the file, you can highlight you code and then click go to Code > Reformat Code.

Reply:
done, thanks!